cognee/api/v1/cognify/cognify.py

```diff
@@ -66,30 +66,50 @@ async def cognify(datasets: Union[str, List[str]] = None, graph_data_model: obje
 
     awaitables = []
 
-    await initialize_graph(USER_ID, graph_data_model)
+    if graph_db_type is None:
+        graph_db_type = GraphDBType.NETWORKX
+
+    graph_client = await get_graph_client(graph_db_type)
+
+    await initialize_graph(USER_ID, graph_data_model, graph_client)
 
     for file_metadata in files_metadata:
         with open(file_metadata["file_path"], "rb") as file:
             file_type = guess_file_type(file)
             text = extract_text_from_file(file, file_type)
 
-            awaitables.append(process_text(text, file_metadata))
+            awaitables.append(process_text(text, file_metadata, graph_data_model, classification_model, summarization_model, labeling_model, graph_model, cognitive_layer_model, graph_db_type))
 
     graphs = await asyncio.gather(*awaitables)
 
     return graphs[0]
 
-async def process_text(input_text: str, file_metadata: dict):
+async def process_text(input_text: str, file_metadata: dict, graph_data_model: object=None, classification_model: object=None, summarization_model: object=None, labeling_model: object=None, graph_model: object=None, cognitive_layer_model: object=None, graph_db_type: object=None):
     print(f"Processing document ({file_metadata['id']})")
 
     classified_categories = []
 
+    if classification_model is None:
+        classification_model= DefaultContentPrediction
+
+    if summarization_model is None:
```

Comment:
This can be done then inside the function that is doing the summarization.

Reply:
removed